packages/components/src/components/CodeEditor.tsx

```diff
@@ -149,41 +151,47 @@ export const CodeEditor = forwardRef<CodeEditorHandle, CodeEditorProps>(
     // Note the `useState` instead of `useMemo`; we never want to recreate EditorState, it would cause bugs.
     // `defaultValue` changes are ignored;
     // `languageSupport` changes (which depends on `project` prop) are handled with `projectRef`.
+
+    const basicExtensions = [
```

Comment:
Old version used a lazy `useState` initializer, so the performance here would be slightly worse.

All these constructor calls are _probably_ cheap, but I'm not sure, so would recommend moving it back inside `useState` callback.